drivers/memc/memc_mcux_flexspi_w956a8mbya.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2023, ithinx GmbH
+ * Copyright (c) 2023, Tonies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ /*
+  * Based on memc_mcux_flexspi_s27ks0641, which is: Copyright 2021 Basalte bv
+  */
+
+#define DT_DRV_COMPAT   nxp_imx_flexspi_w956a8mbya
+
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+
+#include "memc_mcux_flexspi.h"
+
+
+/*
+ * NOTE: If CONFIG_FLASH_MCUX_FLEXSPI_XIP is selected, Any external functions
+ * called while interacting with the flexspi MUST be relocated to SRAM or ITCM
+ * at runtime, so that the chip does not access the flexspi to read program
+ * instructions while it is being written to
+ */
+#if defined(CONFIG_FLASH_MCUX_FLEXSPI_XIP) && (CONFIG_MEMC_LOG_LEVEL > 0)
+#warning "Enabling memc driver logging and XIP mode simultaneously can cause \
```

Comment:
I would drop this `#warning` from the driver- the issue this is intended to highlight is cases where the FlexSPI is being used for XIP, and logging is enabled. In these cases, log functions being called from critical code sections (while the flash is being written to) introduces a read while write hazard.

Since the w956a8mbya device is intended for use as RAM, I don't expect XIP mode to be used here.

Reply:
Hello Daniel, 

you're right, this should only apply to flash devices. I force pushed an update.